java/code/src/com/redhat/rhn/common/util/FileUtils.java

```diff
@@ -130,27 +113,19 @@
      * @return String containing file.
      */
     public static String readStringFromFile(String path, boolean noLog) {
-        if (log.isDebugEnabled()) {
-            log.debug("readStringFromFile: {}", StringUtil.sanitizeLogInput(path));
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("readStringFromFile: {}", StringUtil.sanitizeLogInput(path));
         }
 
-        File f = new File(path);
-        BufferedReader input;
         try {
-            input = new BufferedReader(new FileReader(f));
-            StringWriter writer = new StringWriter();
-            IOUtils.getInstance().copyWriter(input, writer);
-            String contents = writer.toString();
-            if (noLog && log.isDebugEnabled()) {
-                log.debug("contents: {}", contents);
+            String contents = Files.readString(Path.of(path));
```

Comment:
@cbosdo This to me seems a false positive. Sonarcloud report a place where "user can craft an HTTP request with malicious content" but from what I see it's not in any way accessible through HTTP request or user input. The only place I see is that can potentially be under user control is `CobblerSnippet` which already checks the path.

Reply:
The problem I have disabling this issue is that it may be called later in a way that is dangerous: `readStringFromFile()` is very generic. Maybe it would be worth adding a check that the file is in an expected folder. How about adding a parameter like `basePath` and check that the file canonical path is within `basePath`. That of course means that we need to go over all the uses to set that base path. There are not that many instances that are not in tests...

Reply:
Yes, makes sense. I will try to address this so that the method is always safe to call. Another option would be to remove the method completely if there are few usages, since `Files.readString()` exists. If we use it directly, we can have the security check near the actual logic where it makes probably more sense. I will see.

Reply:
I added another version of the method that allows to specify the base directory and the actual file.